### PR TITLE
Fix path related issues in windows with 'Path.toUri()' method

### LINKED ISSUE #41995: [Task]: Validate recent profiler related changes in windows
### Description

$title

### Describe your task(s)

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #41995

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java [hunk 1]
```diff
@@ -136,10 +136,11 @@ private Path getTargetPath(Project project) {
 
     private String getTargetFilePath(Project project) {
         if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
-            return Paths.get(System.getProperty(USER_DIR)).resolve(
-                    getFileNameWithoutExtension(project.sourceRoot()) + BLANG_COMPILED_JAR_EXT).toUri().getPath();
+            return Path.of(Paths.get(System.getProperty(USER_DIR)).resolve(
```
Comment (warunalakshitha) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java:139]: Can we make Paths.get(System.getProperty(USER_DIR))  constant?

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java [hunk 2]
```diff
@@ -52,6 +52,7 @@ public class RunProfilerTask implements Task {
     private final PrintStream err;
     private static final String JAVA_OPTS = "JAVA_OPTS";
     private static final String CURRENT_DIR_KEY = "current.dir";
+    private final Path targetOutputPath = Paths.get(System.getProperty(USER_DIR));
```
Comment (warunalakshitha) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java:55]: Shall we make this constant?
private static final Path TARGET_OUTPUT_PATH = Paths.get(System.getProperty(USER_DIR));

**Review by HindujaB — APPROVED** [review]

**Pull request merged**